crypto/ec/ecx_meth.c

```diff
@@ -92,15 +92,19 @@ static int ecx_key_op(EVP_PKEY *pkey, int id, const X509_ALGOR *palg,
             X25519_public_from_private(pubkey, privkey);
             break;
         case EVP_PKEY_ED25519:
-            ED25519_public_from_private(pubkey, privkey);
+            /*
+             * TODO(3.0): We set the library context to NULL for now. This will
+             * need to change or this code be removed.
+             */
+            ED25519_public_from_private(NULL, pubkey, privkey);
```

Comment:
Once #11494 is merged the libctx will be available and can be used here. Please check the return code and return an error if it fails

Reply:
I've added the test and anticipate a merge conflict to fix.